src/rules/no-restricted-vi-methods.test.ts

```diff
@@ -0,0 +1,71 @@
+import { RuleTester } from '@typescript-eslint/utils/dist/ts-eslint'
+import { it } from 'vitest'
+import rule, { RULE_NAME } from './no-restricted-vi-methods'
+
+it(RULE_NAME, () => {
+	const ruleTester: RuleTester = new RuleTester({
+		parser: require.resolve('@typescript-eslint/parser')
+	})
+	ruleTester.run(RULE_NAME, rule, {
+		valid: [
+			'vi',
+			'vi()',
+			'vi.mock()',
+			'expect(a).rejects;',
+			'expect(a);',
+			{
+				code: `
+					import { vi } from 'vitest';
+					vi;
+				`,
+				parserOptions: { sourceType: 'module' }
+			}
+		],
+
+		invalid: [
+			{
+				code: 'vi.fn()',
+				options: [{ fn: null }],
+				errors: [{
+					messageId: 'restrictedViMethod',
+					data: {
+						message: null,
+						restriction: 'fn'
+					},
+					column: 4,
+					line: 1
+				}]
+			},
```

Comment:
I tried adding the following test, but it kept failing (it wouldn't find any errors). I'm not sure why this test doesn't work:

```
{
	code: 'vi["fn"]()',
	options: [{ fn: null }],
	errors: [{
		messageId: 'restrictedViMethod',
		data: {
			message: null,
			restriction: 'fn'
		},
		column: 4,
		line: 1
	}]
}
```